### PR TITLE
Update Examples Documentation

### PR DESCRIPTION
**Pull request by Lnaden**
Create full documentation for all examples on the [YANK Examples](https://github.com/choderalab/yank-examples) repo.

Proofreading and comments appreciated!

### CODE REVIEW

File: docs/examples/all-ligand-explicit.rst [hunk 1]
```diff
@@ -0,0 +1,184 @@
+.. _all-ligand-explicit:
+
+Absolute Binding of Binders and Non-binders to T4 lysozyme L99A in Explicit Solvent with Combinatorial Options
+==============================================================================================================
+
+This example looks at the absolute binding of a series of binders and non-binder to the T4 Lysozyme L99A protein.
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: "looks at the absolute binding of a series of binders and non-binder" => "performs absolute binding free energy calculations for a series of small molecules that contain known binders and nonbinders"

File: docs/examples/all-ligand-explicit.rst [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+We take advantage of three features of YANK in this example we have not done in others thus far:
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: Let's not assume anyone will work through tutorials serially! Everyone will just dive in directly to the tutorial that is closest to the one they are interested in. If you reference a simpler tutorial, link to it and provide a simple description.

File: docs/examples/all-ligand-explicit.rst [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+1. YANK's ability to build out small molecules from SMILES strings (through OpenEye Toolkit)
+2. YANK's ability to run multiple ligands through the same commands
+3. YANK's ability to run ``Combinatorial`` options.
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: Link to the docs description of `Combinatorial`?

File: docs/examples/all-ligand-explicit.rst [hunk 4]
```diff
@@ -0,0 +1,184 @@
+.. _all-ligand-explicit:
+
+Absolute Binding of Binders and Non-binders to T4 lysozyme L99A in Explicit Solvent with Combinatorial Options
+==============================================================================================================
+
+This example looks at the absolute binding of a series of binders and non-binder to the T4 Lysozyme L99A protein.
+
+We take advantage of three features of YANK in this example we have not done in others thus far:
+
+1. YANK's ability to build out small molecules from SMILES strings (through OpenEye Toolkit)
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: Link to the OpenEye toolkit installation instructions?

File: docs/examples/all-ligand-explicit.rst [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+2. YANK's ability to run multiple ligands through the same commands
+3. YANK's ability to run ``Combinatorial`` options.
+
+This example resides in ``{PYTHON SOURCE DIR}/share/binding/t4-lysozyme``.
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: This should be `$MINICONDA/share/yank/examples/binding/T4-lysozyme-L99A`

File: docs/examples/all-ligand-explicit.rst [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+to
+
+.. code-block:: yaml
+
+    binders:
+        select: <Integer>
+    non-binders:
+        select: <Integer>
+
+where ``<Integer>`` is a single integer representing a line in the corresponding file.
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: You can also use a small list, right?

File: docs/examples/all-ligand-explicit.rst [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+YANK automatically generates the instructions that ``yank analyze`` will use to compute the free energy difference
+for every combination of options. Right now YANK will only tell you the free energy for each individual simulation.
+It will be up to you to trap this information and split it into each simulation.
+
+Future versions of YANK will generate more helpful output for ``!Combinatorial`` simulations.
+
+Other Files in this Example
+---------------------------
+
+We also provide inputs for running implicit simulation of the same problem.
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: List the filenames?

File: docs/examples/freesolv-imp-exp.rst [hunk 8]
```diff
@@ -0,0 +1,67 @@
+.. _freesolv-imp-exp:
+
+Hydration Free Energy of part of the FreeSolv Database in Implicit and Explicit Solvent
```
Comment (jchodera) [inline docs/examples/freesolv-imp-exp.rst]: Maybe "Calculation of absolute hydration free energies from the FreeSolv Database"?

File: docs/examples/freesolv-imp-exp.rst [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+=======================================================================================
+
+This example looks at a the hydration free energy of a subset of the
```
Comment (jchodera) [inline docs/examples/freesolv-imp-exp.rst]: "computes absolute hydration free energies of..."

File: docs/examples/freesolv-imp-exp.rst [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+`FreeSolv <http://link.springer.com/article/10.1007%2Fs10822-014-9747-x>`_
+`database <https://github.com/MobleyLab/FreeSolv>`_ in both implicit and explicit solvent.
+
+This example showcases how easy it is to compute the free energy of a large dataset from a single input file, and how
```
Comment (jchodera) [inline docs/examples/freesolv-imp-exp.rst]: "to set up and run free energy calculations for a large dataset"

File: docs/examples/host-guest-implicit.rst [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+.. code-block:: yaml
+
+   options:
+     minimize: yes
+     verbose: yes
+     number_of_iterations: 500
+     temperature: 300*kelvin
+     restraint_type: FlatBottom
+     pressure: null
```
Comment (jchodera) [inline docs/examples/host-guest-implicit.rst]: If we omit `pressure`, it is automatically `null`, right?

File: docs/examples/host-guest-implicit.rst [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+We set ``implicit_solvent`` to tell YANK to actually a continuum solvent. If we had not set this, we would have gotten
+a solvent representing a vacuum. The argument given to ``implicit_solvent`` is linked to OpenMM's implicit solvent names.
+In this case, the ``OBC2`` name is the Onufriev-Bashford-Case GBSA model. We could change the dielectric, but YANK
+defaults to the dielectric for TIP3P water.
+
+
+Reamining Headings
+^^^^^^^^^^^^^^^^^^
+
+The remaining headings: ``systems``, ``protocols``, and ``experiments`` are strait forward. Please see the more
```
Comment (jchodera) [inline docs/examples/host-guest-implicit.rst]: "straightforward"

File: docs/examples/hydration-phenol-explicit.rst [hunk 13]
```diff
@@ -0,0 +1,127 @@
+.. _hydration_phenol_explicit:
+
+Hydration Free Energy of Phenol in Explicit TIP3P Water
```
Comment (jchodera) [inline docs/examples/hydration-phenol-explicit.rst]: Is this example really separate? It's not just a subset of the FreeSolv dataset?

File: docs/examples/all-ligand-explicit.rst [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+This example performs absolute binding free energy calculations for a series of small molecules that contain known
+binders and nonbinders to the T4 Lysozyme L99A protein.
+
+We take advantage of three advanced features of YANK in this example (please see the
+:doc:`basic tutorial for the YAML sections and binding free energy calculation <p-xylene-explicit>`
+for learning how to set up basic YANK simulations):
+
+1. YANK's ability to build out small molecules from SMILES strings (through OpenEye Toolkit)
+
+   * Please see the `OpenEye Install Instructions <http://docs.eyesopen.com/toolkits/python/quickstart-python/install.html>`_ for installing these tool kits
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: We have some instructions we should possibly link to:
http://getyank.org/latest/installation.html#optional-tools

File: docs/examples/all-ligand-explicit.rst [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+
+2. YANK's ability to run multiple ligands through the same commands
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: You might link to the [molecules](http://getyank.org/latest/yamlpages/molecules.html) section of the YAML description

File: docs/examples/all-ligand-explicit.rst [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+3. YANK's ability to run ``Combinatorial`` options
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: You might link to the [Combinatorial](http://getyank.org/latest/yamlpages/combinatorial.html) section of the YAML docs.

File: docs/examples/all-ligand-explicit.rst [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+      binders:
+        filepath: input/L99A-binders.csv
+        antechamber:
+          charge_method: bcc
+        select: all
+      non-binders:
+        filepath: input/L99A-non-binders.csv
+        antechamber:
+          charge_method: bcc
+        select: !Combinatorial [4,5,6,7,10,11,12,13,14,15,18,19,28,29,30,31,32,33,34,35,36,37,41,42,43,48,49,50,56,58,59]
```
Comment (jchodera) [inline docs/examples/all-ligand-explicit.rst]: This is a weird subset of ligands, isn't it? Did you just exclude ligands we can't deal with?

File: docs/examples/host-guest-implicit.rst [hunk 18]
```diff
@@ -0,0 +1,125 @@
+.. _host_guest_implicit:
+
+Absolute Binding of Host cucurbit[7]uril with Guest molecule B2 in Implicit Solvent
+===================================================================================
+
+This example illustrates the computation of the binding free energy of B2 guest to the host cucurbit[7]uril in an
```
Comment (jchodera) [inline docs/examples/host-guest-implicit.rst]: We should link to [this AMBER tutorial](http://ambermd.org/tutorials/advanced/tutorial21/) that was the source of files for CB7 and B2.

**Comment by jchodera** [comment]
I think this just has a few fixes to be made and then can be merged!

**Comment by Lnaden** [comment]
There will probably be some extra touch ups once we get the other PRs in, but we can fix all that in an additional pr

**Pull request merged**